src/AlgebraicPetri.jl

```diff
@@ -213,6 +213,31 @@ inputs(p::AbstractPetriNet, t) = subpart(p, incident(p, t, :it), :is)
 """
 outputs(p::AbstractPetriNet, t) = subpart(p, incident(p, t, :ot), :os)
 
+
+""" induced_subnet(p::T, t)
+
+Return the induced subnet consisting of the transition(s) in `t`, all input
+places and input arcs, and all output places and output arcs. The returned
+object will be of the same type as `p`.
+"""
+function induced_subnet(p::T, t) where {T <: AbstractPetriNet}
+  induced_subnet(p, [t])
+end
+
+function induced_subnet(p::T, t::AbstractVector{Int}) where {T <: AbstractPetriNet}
+  @assert all([tt ∈ parts(p,:T) for tt in t])
+
+  input_arcs = incident(p, t, :it)
+  output_arcs = incident(p, t, :ot)
+
+  input_places = [subpart(p, input_i, :is) for input_i in input_arcs]
+  output_places = [subpart(p, output_i, :os) for output_i in output_arcs]
+
+  subnet = T()
+  copy_parts!(subnet, p, T=t, I=vcat(input_arcs...), O=vcat(output_arcs...), S=union(input_places..., output_places...))
```

Comment:
@epatters, this is done with the subpart and incident API. Would you prefer if this used the Topos logic operations? Is not(negation(tt)) equivalent to this formulation?